packages/cloud_firestore/cloud_firestore/lib/src/query.dart

```diff
@@ -344,28 +341,28 @@ class Query {
     dynamic isGreaterThan,
     dynamic isGreaterThanOrEqualTo,
     dynamic arrayContains,
-    List<dynamic> /*?*/ arrayContainsAny,
-    List<dynamic> /*?*/ whereIn,
-    List<dynamic> /*?*/ whereNotIn,
-    bool /*?*/ isNull,
+    List<dynamic>? arrayContainsAny,
+    List<dynamic>? whereIn,
+    List<dynamic>? whereNotIn,
+    bool? isNull,
   }) {
     _assertValidFieldType(field);
 
-    final ListEquality<dynamic> equality = const ListEquality<dynamic>();
+    const ListEquality<dynamic> equality = ListEquality<dynamic>();
     final List<List<dynamic>> conditions =
         List<List<dynamic>>.from(parameters['where']);
 
     // Conditions can be chained from other [Query] instances
     void addCondition(dynamic field, String operator, dynamic value) {
       List<dynamic> condition;
-      value = _CodecUtility.valueEncode(value);
+      dynamic codecValue = _CodecUtility.valueEncode(value);
```

Comment:
Why dynamic?

Reply:
It was overriding the param `value` which is itself `dynamic`. The method itself returns `dynamic`

Reply:
Can either `field` or `value` be null? If not then we should use `Object` instead for non-null as 'dynamic' values can accept null. I think this is what Remi might be referring to

Reply:
It's more that I quite dislike the `dynamic` keyword (and would use `Object` or `Object?` instead), but yeah that's a fair question.

Reply:
It's not blocking anyway. We can merge this without making change here